onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -663,6 +663,14 @@ static void RegisterExecutionProviders(InferenceSession* sess, const std::vector
     } else if (type == kRknpuExecutionProvider) {
 #ifdef USE_RKNPU
       RegisterExecutionProvider(sess, *onnxruntime::CreateExecutionProviderFactory_Rknpu());
+#endif
+    } else if (type == kCoreMLExecutionProvider) {
+#if defined(USE_COREML)
+#include <TargetConditionals.h>
+#if !TARGET_OS_IPHONE
+      LOGS_DEFAULT(WARNING) << "CoreML execution provider can only be used to generate ORT format model in this build.";
+#endif
```

Comment:
For now we don't need these line, we can add these when we have the CoreML EP ready for converting model script on non-Mac platform